app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.kt

```diff
@@ -159,39 +164,38 @@ object EventPlatformClient {
             }
         }
 
-        @SuppressLint("CheckResult")
         private fun sendEventsForStream(streamConfig: StreamConfig, events: List<Event>) {
-            (if (ReleaseUtil.isDevRelease)
-                ServiceFactory.getAnalyticsRest(streamConfig).postEvents(events)
-            else
-                ServiceFactory.getAnalyticsRest(streamConfig).postEventsHasty(events))
-                    .subscribeOn(Schedulers.io())
-                    .subscribe({
-                        when (it.code()) {
-                            HttpURLConnection.HTTP_CREATED,
-                            HttpURLConnection.HTTP_ACCEPTED -> {}
-                            else -> {
-                                // Received successful response, but unexpected HTTP code.
-                                // TODO: queue up to retry?
-                            }
-                        }
-                    }) {
-                        L.e(it)
-                        if (it is HttpStatusException) {
-                            if (it.code >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
-                                // TODO: For errors >= 500, queue up to retry?
-                            } else {
-                                // Something unexpected happened.
-                                if (ReleaseUtil.isDevRelease) {
-                                    // If it's a pre-beta release, show a loud toast to signal that
-                                    // a potential issue should be investigated.
-                                    WikipediaApp.instance.mainThreadHandler.post {
-                                        Toast.makeText(WikipediaApp.instance, it.message, Toast.LENGTH_LONG).show()
-                                    }
-                                }
+            if (!WikipediaApp.instance.isOnline) {
```

Comment:
This check is technically not necessary. In the `sendAllScheduled()` function, it checks `if (ENABLED)`, and the variable `ENABLED` is set automatically by the `ConnectionStateMonitor` which detects whenever the app goes offline and online.

Reply:
Removed.